diff --git a/extendedcommands.c b/extendedcommands.c
old mode 100644
new mode 100755
index 6db4541..bc29e16
--- a/extendedcommands.c
+++ b/extendedcommands.c
@@ -1217,7 +1217,7 @@ int is_path_mounted(const char* path) {
 }
 
 int has_datadata() {
-    Volume *vol = volume_for_path("/datadata");
+    Volume *vol = volume_for_path("/userdata");
     return vol != NULL;
 }
 
diff --git a/nandroid.c b/nandroid.c
old mode 100644
new mode 100755
index 315660c..4d5de70
--- a/nandroid.c
+++ b/nandroid.c
@@ -257,6 +257,9 @@ int nandroid_backup(const char* backup_path)
     if (0 != (ret = nandroid_backup_partition(backup_path, "/recovery")))
         return ret;
 
+    if (0 != (ret = nandroid_backup_partition(backup_path, "/kernel")))
+        return ret;
+
     Volume *vol = volume_for_path("/wimax");
     if (vol != NULL && 0 == stat(vol->device, &s))
     {
@@ -532,6 +535,8 @@ int nandroid_restore(const char* backup_path, int restore_boot, int restore_syst
     if (restore_boot && NULL != volume_for_path("/boot") && 0 != (ret = nandroid_restore_partition(backup_path, "/boot")))
         return ret;
     
+    if (NULL != volume_for_path("/kernel") && 0 != (ret = nandroid_restore_partition(backup_path, "/kernel")))
+        return ret;
     struct stat s;
     Volume *vol = volume_for_path("/wimax");
     if (restore_wimax && vol != NULL && 0 == stat(vol->device, &s))
